WebPhapp/backend/server.js

```diff
@@ -566,15 +495,237 @@ app.get('/api/v1/patients/:patientID', (req,res) => {
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/dispensers/prescriptions/historical/1"
+    To be used in Axois call:
+        .get("/api/v1/dispensers/prescriptions/historical/1")
+Returns:
+    list<Prescription>
+*/
+app.get('/api/v1/dispensers/prescriptions/historical/:dispenserID', (req, res) => {
+    var dispenserID = parseInt(req.params.dispenserID);
+    var handlePrescriptionsCallback = function(prescriptions) {
+        // only take historical prescriptions with dispenserID
+        prescriptions = prescriptions.filter(
+            prescription =>
+                prescription.dispenserID === dispenserID &&
+                (prescription.cancelDate > 0 || prescription.refillsLeft < 1)
+        );
+
+        // Convert date integers to strings
+        prescriptions = prescriptions.map(
+            prescription => convertDatesToString(prescription)
+        );
+
+        console.log('Sending ' + prescriptions.length.toString()
+                        + ' historical prescription(s) related to dispenserID ' + dispenserID.toString());
+        res.status(200).send(prescriptions);
+    }
+    
+    // Error if dispenser ID is null or undefined
+    if(dispenserID == null) {
+        console.log('/api/v1/dispensers/prescriptions/historical/:dispenserID: No ID match');
+        res.status(400).send([]);
+        return;
+    }
+
+    if(conn.Blockchain) {
+        var field_dispenserID = 2;
+        block_helper.read_by_value(field_dispenserID, dispenserID)
+        .then((answer) => {
+            handlePrescriptionsCallback(answer.prescriptions);
+        })
+        .catch((error) => {
+            console.log('error: ', error);
+            res.status(400).send('Error in searching blockchain for prescriptions matching dispenserID.');
+        });
+    }
+    else { // search from dummy data
+        var prescriptions = readJsonFileSync(
+            __dirname + '/' + "dummy_data/prescriptions.json").prescriptions;
+        handlePrescriptionsCallback(prescriptions);
+    }
+});
+
+/*
+About:
+    An api endpoint that returns all open prescriptions for a given dispenserID.
+    A prescription is open if it can be fulfilled (not cancelled and can be refilled).
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/dispensers/prescriptions/open/1"
+    To be used in Axois call:
+        .get("/api/v1/dispensers/prescriptions/open/1")
+Returns:
+    list<Prescription>
+*/
+app.get('/api/v1/dispensers/prescriptions/open/:dispenserID', (req, res) => {
+    var dispenserID = parseInt(req.params.dispenserID);
+    var handlePrescriptionsCallback = function(prescriptions) {
+        // only take open prescriptions with dispenserID
+        prescriptions = prescriptions.filter(
+            prescription => 
+                prescription.dispenserID === dispenserID
+                && prescription.refillsLeft > 0
```

Comment:
I do have a slight issue with this logic after reviewing the prescription fields. I remember we discussed the different attributes on the blockchain with the `daysValid` field. One interpretation of this was to have this be the amount of days before someone can get a refill. The other was the amount of total days the prescription is valid for. We need both of these values on a prescription.

Reply:
In the future, the once the `daysBetween` is in the fields that we need to use, then we'll need to implement a check that prescriptions can only be redeemed if the `daysBetween`. Should not change the implementation here, but it will massively change it on the frontend and the redemption process.